src/components/BottomNavigation/index.tsx

```diff
@@ -1,16 +1,21 @@
 import { BottomNavigationStyled, Groups, Profile, AddExpense } from './index.styled';
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useParams } from 'react-router-dom';
 import groups from '../../assets/images/groups-icon.svg'
 import receipt from '../../assets/images/receipt-icon.svg'
 import avatar from '../../assets/images/avatar-icon.svg'
-import { ROUTES } from '../../constants/routes';
+import { CreateRouteString, ExpenseRouteMode, NoneRouteString, ROUTES, composeExpenseRoute } from 'constants/routes';
 
 interface Props {
   visible?: boolean;
 }
 
 const BottomNavigation = ({ visible = true }: Props) => {
   const navigate = useNavigate();
+  const { groupId } = useParams();
+
+  const onAddExpenseClick = () => {
+    navigate(composeExpenseRoute(groupId ?? NoneRouteString, CreateRouteString, ExpenseRouteMode.ADD_MAININFO));
```

Comment:
Looks very strange and unclear. May be create some factories for routes? 
Example: 
ExpenseRouteFactory
- createAddMainInfoRoute(groupId)
- createListRoute(groupId)
...

Reply:
It is just a factory function that accepts some arguments, why is it strange/unclear?